client/runnerservice.go

```
@@ -0,0 +1,195 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"text/tabwriter"
+	"time"
+
+	"github.com/NajiObeid/teleport/transport/go"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+const (
+	contextTimeout = 10 * time.Second
+
+	clientCertFile   = "cert/client.pem"
+	clientKeyFile    = "cert/client-key.pem"
+	serverCACertFile = "cert/ca-cert.pem"
+)
+
+type runnerService struct {
+	client transport.RunnerServiceClient
+}
+
+func NewRunnerService(port string) (*runnerService, error) {
+	tlsCredentials, err := loadTLSCredentials()
+	if err != nil {
+		return nil, err
+	}
+
+	grpcConnection, err := grpc.Dial(":"+port, grpc.WithTransportCredentials(tlsCredentials))
+	if err != nil {
+		return nil, err
+	}
+
+	return &runnerService{
+		client: transport.NewRunnerServiceClient(grpcConnection),
+	}, nil
+}
+
+func (rs *runnerService) Run(command string) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(contextTimeout))
+	defer cancel()
+
+	response, err := rs.client.Run(
+		ctx,
+		&transport.RunRequest{
+			Command: command,
+		},
+	)
+	if err != nil {
+		fmt.Printf("error starting job: %+v\n", err)
+		return
```

Why not return the error to the caller?